rust/core-lib/src/view.rs

```diff
@@ -233,6 +233,11 @@ impl View {
         &self.selection
     }
 
+    /// Collapse all selections in this view into a single caret
```

Comment:
Pretty sure you'll need to set sel_dirty here one way or another. Setting it directly would work on current master, but that won't merge cleanly with #441.

What I'd recommend is cloning the selection, collapsing your clone, and calling set_selection. This will let minimal invalidation compute a good diff, for which it needs both the "before" and "after" values of the selection.